src/components/settings/SettingsSSOTab/index.jsx

```diff
@@ -22,9 +25,13 @@ const SettingsSSOTab = ({ enterpriseId }) => {
         </Hyperlink>
       </div>
       {!isLoading && (
-        <p>
-          {existingConfigs.length > 0 && <ExistingSSOConfigs />}
+        <div>
+          {/* Configs found and editing mode is off, so let's go to listing page */}
+          {existingConfigs.length > 0 && (providerConfig === null) && <ExistingSSOConfigs configs={existingConfigs} />}
```

Comment:
yeah I know it could be confusing. Basically, `provideConfig` here is a chosen config. Such as after clicking on a listed config to edit it or view it. If nothing has been chosen yet, but there are existing configs to be shown, then take to existing listing page. If a config was chosen, just take to the NewSSOForm which also doubles as editing flow